### PR TITLE
Add upload preview

### PR DESCRIPTION
**Pull request by sibinhho99**
([#32](`https://github.com/WING-NUS/SSID/issues/32`))
Screenshot 2022-07-11 at 4 43 50 PM
Screenshot 2022-07-11 at 4 43 39 PM
Screenshot 2022-07-11 at 4 44 00 PM

### CODE REVIEW

File: app/views/assignments/_form.html.erb [hunk 1]
```diff
@@ -11,61 +11,48 @@
   <% end %>
   <%= hidden_field_tag "which_form", (@assignment.id ? "update_assignment" : "new_assignment"), { :id => "which-form" } %>
   <div class="row">
-  <p> Hover over ⓘ for more details about any terminology.</p>
```
Comment (knmnyn) [inline app/views/assignments/_form.html.erb:14]: Not sure why diff generates this block as a difference in code.  It looks identical.

File: app/views/assignments/_form.html.erb [hunk 2]
```diff
@@ -11,61 +11,48 @@
...
-    if ($('#which-form').val() === "update_assignment") {
-      document.querySelector('#assignment_title').disabled = true;
-      document.querySelector('#assignment_language').disabled = true;
-      document.querySelector('#assignment_min_match_length').disabled = true;
-      document.querySelector('#assignment_ngram_size').disabled = true;
-    }
-  })
-</script>
+<%= javascript_include_tag "assignments_file_upload" %>
+<%= javascript_include_tag "jszip.min" %>
```
Comment (knmnyn) [inline app/views/assignments/_form.html.erb:57]: You should document somewhere in README.md that the jszip.min comes from where.

Comment (knmnyn) [reply app/views/assignments/_form.html.erb:57]: And its MD5 hash so that the source has a known version.

**Review by knmnyn — COMMENTED** [review]
Generally looks ok, but still needs work on documentation.

File: app/assets/javascripts/assignments_file_upload.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with SSID.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+var checkbox = document.getElementById('assignment_mapbox');
+var details_div = document.getElementById('map_file_details');
+checkbox.onchange = function () {
```
Comment (nguyen-mai-huong) [inline app/assets/javascripts/assignments_file_upload.js:21]: The 3-dot menu site-wide is not working e.g. cannot access Edit/Delete a course ...

**Comment by nguyen-mai-huong** [comment]
1. It's a good idea to organize code in a cleaner way. Just that we need to be a bit careful when doing it.
Currently, assignments_file_upload is globally accessible. These functions should be available to Assignment only.

2. Another note is that it's good to organize jszip.min.js into a library folder like /javascripts/jit or /javascripts/prettify.
It will help us keep track of the external JS libs we are using, and let us know when we need to upgrade JS dependencies management when the number of libs grows.

File: app/assets/javascripts/assignments_file_upload.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    console.log($('which-form').val());
+    if ($('#which-form').val() === "update_assignment") {
+        document.querySelector('#assignment_title').disabled = true;
+        document.querySelector('#assignment_language').disabled = true;
+        document.querySelector('#assignment_min_match_length').disabled = true;
+        document.querySelector('#assignment_ngram_size').disabled = true;
+    }
+})
+
+$("#assignment_file").attr("onchange", "onAssignmentFileInputChange(this)");
```
Comment (nguyen-mai-huong) [inline app/assets/javascripts/assignments_file_upload.js:38]: Not recommended way of binding an event. You can try
`        $("#assignment_file").on("change", function() {
          Assignment.changeAssignmentFileInput(this);
        });`

**Review by nguyen-mai-huong — CHANGES_REQUESTED** [review]

File: app/views/assignments/_form.html.erb (hunk 2 above)
Comment (knmnyn) [reply app/views/assignments/_form.html.erb:57]: Yes, that is sufficient.  There should be documentation in the README.md or subcomponent README that documents such dependencies, much like what a Gemfile does.

**Pull request merged**